### PR TITLE
Documentation

### LINKED ISSUE #3: Documentation
- Update README
- Add [Yard](https://yardoc.org/)?
- Document `CwTranslate::Cache` and `CwTranslate::Translator` APIs
- Document `CwTranslate` API

### PR DESCRIPTION
**Pull request by AlaaSarhan**
Closes #3 

- [x] Update README
- [x] Add [Yard](https://yardoc.org/)?
- [x] Document `CwTranslate::Cache` and `CwTranslate::Translator` APIs
- [x] Document `CwTranslate` API
- [x] Move configuration to initialization level
- [x] Introduce & document `CwTranslate::Translate` class
- [x] Introduce & document `CwTranslate::Config` class
- [x] Update specs accord. changes
- [x] Add specs for updated CwTranslate::Translate

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -3,6 +3,5 @@ set -euo pipefail
 IFS=$'\n\t'
 set -vx
 
-bundle install
+bundle install --path vendor/bundle
```
Comment (irmela) [inline bin/setup]: Just curious about what the advantage of `--path vendor/bundle` is in this case?

File: lib/cw_translate/translate.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      translation(text, no_cache).tap do |translation|
+        use_cache && @cache.updateCache(text, to, opts[:from], translation)
+      end
+    end
+
+    private
+
+    # @private
+    def translation(text, use_cache)
+      use_cache && cache.lookupCache(text, to, opts[:from]) ||
```
Comment (irmela) [inline lib/cw_translate/translate.rb]: Do you think we can find a more readable solution here? For example:

```ruby
def translation(text, use_cache)
  return cache.lookupCache(text, to, opts[:from]) if use_cache
  @translator.translate(text, to, opts[:from])
end
```

**Review by irmela — COMMENTED** [review]
Looks very good! 🚀 
Except that test are failing (but there already is a ToDo item for that) I only have two minor comments/questions.

File: bin/setup (hunk 1 above)
Comment (AlaaSarhan) [reply bin/setup]: gonna remove this and fix the problem I needed it for on my machine

File: lib/cw_translate/translate.rb (hunk 2 above)
Comment (AlaaSarhan) [reply lib/cw_translate/translate.rb]: Changed this, thought cannot return only if `use_cache` is true. If cache doesn't contain it yet (first hit) it should proceed and use translate on next line ;)

**Comment by AlaaSarhan** [comment]
@irmela hopefully it will be the biggest PR in this repo ;)

**Review by irmela — APPROVED** [review]
🚀

**Pull request merged**